tests/utils.py

```diff
@@ -17,3 +20,13 @@ def table_record_is_valid(table_record: RecordType,
                and column.nullable
                or isinstance(coordinate, column.type.python_type)
                for coordinate, column in zip(table_record, table.columns))
+
+def records_satisfy_table_constraints(records: List[RecordType],
+                                      *,
+                                      table: Table) -> bool:
+    unique_indices = [index
+                      for index, column in enumerate(table.columns)
+                      if is_column_unique(column)]
+    # All returned values are unique for the columns marked as unique
+    return all(len(set(record[index] for record in records)) == len(records)
+               for index in unique_indices)
```

Comment:
I'm thinking about how to make this code self-explanatory (and delete comment): we can collapse these 2 comprehensions in one (more complex, but straightforward IMO):
```python
    return all(len(set(record[index] for record in records)) == len(records)
               for index, column in enumerate(table.columns)
               if is_column_unique(column))
```

Reply:
further we can write an utility for checking if iterable contains unique elements like
```python
def all_unique(iterable: Iterable[Hashable]) -> bool:
    seen = set()
    for element in iterable:
        if element in seen:
            return False
        seen.add(element)
    return True
```
(looks like it should work)
and after that use it here
```python
all(all_unique(record[index] for record in records)
    for index, column in enumerate(table.columns)
    if is_column_unique(column))
```

Reply:
what do you think?

Reply:
I think the second option is more readable, and wrapping it into one comprehension is the way to go. Will implement it!